File: tests/fixtures/v2-deserialization-flow/positive/DeserializationFlow.cs
using System.Runtime.Serialization.Formatters.Binary;
using System.Text.Json;

class DeserializationFlow
{
    object Direct(dynamic Request)
    {
        return new BinaryFormatter().Deserialize(Request.Form["payload"]);
    }

    object Propagated(dynamic Request)
    {
        var serialized = Request.Form["payload"];
        var forwardedValue = serialized;
        return new BinaryFormatter().Deserialize(forwardedValue);
    }

    UploadDto Restricted(dynamic Request)
    {
        return JsonSerializer.Deserialize<UploadDto>(Request.Form["restricted"]);
    }
}

File: tests/fixtures/v2-deserialization-flow/positive/DeserializationFlow.java
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.InputStream;
import java.io.ObjectInputStream;

class DeserializationFlow {
    Object direct(HttpServletRequest request) throws Exception {
        return new ObjectInputStream(request.getInputStream()).readObject();
    }

    Object propagated(HttpServletRequest request) throws Exception {
        InputStream serialized = request.getInputStream();
        InputStream alias = serialized;
        return new ObjectInputStream(alias).readObject();
    }

    UploadDto restricted(HttpServletRequest request, ObjectMapper mapper) throws Exception {
        return mapper.readValue(request.getParameter("restricted"), UploadDto.class);
    }
}
